src/pandablocks/asyncio.py

```diff
@@ -151,29 +173,79 @@ def raise_timeouterror():
             raise asyncio.TimeoutError(f"No data received for {frame_timeout}s")
             yield
 
-        async def periodic_flush():
-            if flush_period is not None:
-                while True:
-                    # Every flush_period seconds flush and queue data
-                    await asyncio.sleep(flush_period)
-                    queue.put_nowait(connection.flush())
+        async def flush_manual():
+            while True:
+                await flush_event.wait()
+                queue.put_nowait(connection.flush())
+                flush_event.clear()
+
+        async def flush_periodic():
+            while True:
+                await asyncio.sleep(flush_period)
+                queue.put_nowait(connection.flush())
+
+        async def flush_immediate():
+            pass
```

Comment:
As per https://github.com/PandABlocks/PandABlocks-ioc/pull/73#discussion_r1396018637
We can support manual flush in periodic mode if we use this function for both MANUAL and PERIODIC:

```suggestion
        async def flush_loop():
            while True:
                try:
                    await asyncio.wait_for(flush_event.wait(), flush_period)
                except asyncio.TimeoutError:
                    pass
                else:                    
                    flush_event.clear()
                queue.put_nowait(connection.flush())                
```

Reply:
As mentioned in the [IOC discussion here](https://github.com/PandABlocks/PandABlocks-ioc/pull/73#discussion_r1396862852), I'm not in favour of blurring the boundaries between the enum modes. If we want to do this I'd advocate for a new FlushMode called "PeriodicAndManual" or somesuch.